src/dds_data.py

```diff
@@ -103,15 +106,45 @@ def remove_domain_participant(self, domain_id: int, participant: DcpsParticipant
                     del self.participants[domain_id][idx]
                     self.removed_participant_signal.emit(domain_id, str(participant.key))
 
+    def check_qos_mismatches(self, domain_id):
+        """Check the given domain if there are reader and writer
+        with a qos mismatch. A signal is emitted on mismatch."""
+
+        if domain_id in self.endpoints.keys():
+            if domain_id in self.mismatches:
+                del self.mismatches[domain_id]
+            for (entity_type, endpoint) in self.endpoints[domain_id]:
+                for (entity_type_iter, endpoint_iter) in self.endpoints[domain_id]:
+                    if (endpoint.topic_name == endpoint_iter.topic_name and
+                            endpoint.key != endpoint_iter.key and
+                            entity_type_iter != entity_type):
+
+                        if entity_type.READER and entity_type_iter.WRITER:
```

Comment:
Wouldn't it be practical to define class structure for where one has `Endpoint` with two subclasses: `Reader` and `Writer`? Then you can use `isinstance` and the like instead of having to carry the `entity_type` around. (The IDL/XTypes/type stuff in the Python binding does that and it seems to work quite nicely.)

I only see this detail now. I think making such a change would affect quite a bit more code. Even if this is a good idea, it means it doesn't make sense to change it as part of add the QoS compatibility checking.

Reply:
thats a good idea but i will do it in the next pull-request. Because i want to change something in the data handling there anyway - that would fit good together with the changes i want to make there next.